### ISSUE TITLE
"Unkown error" on publish

### ISSUE THREAD

**Issue by VACEfron** [WilliamVenner/gmpublisher#36]
Hi, when pressing the "Publish!" button, the program displays a green "publish addon" (or something similar) popup, quickly followed by a red popup saying "Unknown error", again followed by the program crashing.

Looking at my workshop items, the program seems to have uploaded an empty addon.

Relevant screenshots below. 

![x7s15](https://user-images.githubusercontent.com/46462862/120871073-f1f60600-c59a-11eb-9609-ff1e4b96a6a0.png)
![h6430](https://user-images.githubusercontent.com/46462862/120871075-f4f0f680-c59a-11eb-8cdc-4259cf1156d9.png)

**Comment by WilliamVenner** [comment]
Does 2.4.0 work?

**Comment by VACEfron** [comment]
No, unfortunately. I've also tried removing spaces from the path to the directory as an attempt to fix it, which also didn't work.

**Comment by VACEfron** [comment]
I've tried some more things. I updated back to v2.5.1 and this time published with an icon which seemed to fix the issue. I'm not too knowledgeable about the Workshop, so I don't know if having an icon is required. This might be a bug in gmpublisher, depending on the answer.

**Comment by WilliamVenner** [comment]
Icon isn't required - it will upload a default one for you if you don't provide one.

**Comment by WilliamVenner** [comment]
Hi, I can't seem to reproduce this problem @VACEfron

Publishing a workshop item with the default icon seems to work fine. Updating in this case also works fine.

What OS are you running, and do you have an antivirus/firewall running? gmpublisher internally uses a websocket to communicate some progress data between the backend and frontend; so it could be being interfered with.

**Comment by VACEfron** [comment]
I'm running Windows 10 version 20H2. I have Windows' built-in defender active.

**Comment by WilliamVenner** [comment]
Can you grab me the debug logs like this?

https://i.venner.io/izKHvVkmJv.mp4

`gmpublisher > output.msg 2> output.err`

**Comment by VACEfron** [comment]
Sure thing.
[output.zip](https://github.com/WilliamVenner/gmpublisher/files/6757440/output.zip)
